File: python/bc1329.py
while True:
    rJo = rMa = 0
    count = int(input())

    if count == 0: break

    cVal = input().split(" ")

    for c in cVal:
        if c == '0': rMa += 1
        else: rJo += 1

    print(f"Mary won {rMa} times and John won {rJo} times")

File: cpp/bc1329.cpp
#include <iostream>
#include <string>

using namespace std;

void calc(int co)
{
    int rJo = 0, rMa = 0, lim = co*2;
    string line;
    
    getline(cin, line);
    
    for(int i = 0; i < lim; i++)
    {
        if(line[i] == '0') rMa++;
        else if(line[i] == '1') rJo++;
    }
    
    cout << "Mary won " << rMa << " times and John won " << rJo << " times" << endl;
    cin.clear();
}

int main()
{
    int count;
    
    while(true)
    {
        cin >> count;
        
        if(count == 0) break;
        else  
        {
            cin.ignore();
            calc(count);
        }    
    }
    
}
